### PR TITLE
omit invalid aws tags in rds autodiscovery

### LINKED ISSUE #9737: Database auto-discovery fails with databases created by CloudFormation.
### Description

**What happened**:
I created a MySql RDS instance via CloudFormation scripts and setup the teleport database auto-discovery agent. The database wasn't showing up in Teleport however. Once I confirmed permissions were correct I turned on debugging and looked at the logs. I saw the following error repeated:

`INFO [RDS-WATCH] Could not convert RDS instance "*******" to database resource: invalid label key: "aws:cloudformation:stack-id". labels:map[*:[*]] region:us-east-1 watchers/rds.go:114`

The root of the problem seems to be with this function: https://github.com/gravitational/teleport/blob/807c6d6627ad13c67ad1e14a8457962a725320df/lib/services/database.go#L219-L226 

It's adding every tag as a label, including ones with reserved characters. Could we adjust the logic to either filter out invalid tags or give us the ability to opt-in only specific tags to be included as labels?

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Create a AWS::RDS::DBInstance in CloudFormation
2. Turn on Teleport Database auto-discovery agent.

### Server Details
- Teleport version (run `teleport version`): 8.0.7
- Server OS (e.g. from `/etc/os-release`): Debian 11
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS

### PR DESCRIPTION
**Pull request by greedy52**
closes #9737

### CODE REVIEW

**Review by jimbishopp — APPROVED** [review]

File: lib/services/database.go [hunk 1]
```diff
@@ -218,11 +218,17 @@ func labelsFromRDSCluster(rdsCluster *rds.DBCluster, meta *types.AWS) map[string
...
 }
 
-// tagsToLabels converts RDS tags to a labels map.
-func tagsToLabels(tags []*rds.Tag) map[string]string {
+// rdsTagsToLabels converts RDS tags to a labels map.
+func rdsTagsToLabels(tags []*rds.Tag) map[string]string {
 	labels := make(map[string]string)
 	for _, tag := range tags {
-		labels[aws.StringValue(tag.Key)] = aws.StringValue(tag.Value)
+		// An AWS tag key has a pattern of "^([\p{L}\p{Z}\p{N}_.:/=+\-@]*)$",
```
Comment (r0mant) [inline lib/services/database.go:225]: Nit: Include link to AWS docs? If there's one.

File: lib/services/database.go [hunk 2]
```diff
@@ -218,11 +218,17 @@ func labelsFromRDSCluster(rdsCluster *rds.DBCluster, meta *types.AWS) map[string
...
+		// which can make invalid labels (for example "aws:cloudformation:stack-id").
+		// Omit those to avoid resource creation failures.
+		key := aws.StringValue(tag.Key)
+		if types.IsValidLabelKey(key) {
+			labels[key] = aws.StringValue(tag.Value)
+		}
```
Comment (r0mant) [inline lib/services/database.go:231]: Nit: I'd log a debug message if we can't take in the key.

**Review by r0mant — APPROVED** [review]

File: lib/services/database.go (hunk 2 above)
Comment (jimbishopp) [reply lib/services/database.go:231]: I was thinking the same thing but there is no logger in scope here except the root logger.

**Comment by r0mant** [comment]
@greedy52 Don't forget to backport to v8 please.

**Pull request merged**